assets/javascripts/discourse/controllers/preferences-saved-searches.js

```diff
@@ -0,0 +1,50 @@
+import Controller from "@ember/controller";
+import { action } from "@ember/object";
+import discourseComputed from "discourse-common/utils/decorators";
+import { ajax } from "discourse/lib/ajax";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+
+export default Controller.extend({
+  saving: false,
+
+  @discourseComputed("savedSearches.length", "siteSettings.max_saved_searches")
+  canAddSavedSearch(length, max) {
+    return length < max;
+  },
+
+  @action
+  addSavedSearch() {
+    this.savedSearches.pushObject({ query: "" });
+  },
+
+  @action
+  removeSavedSearch(savedSearch) {
+    this.savedSearches.removeObject(savedSearch);
+  },
+
+  @action
+  save() {
+    this.setProperties({ saved: false, isSaving: true });
+
+    const savedSearches = this.savedSearches
+      .map((savedSearch) => {
+        const query = savedSearch.query.trim();
```

Comment:
could this code just be:

```javascript
.map((savedSearch) => savedSearch?.query?.trim())
```

?

trim can't return null/undefined afaik, and if you call trim on null/undefined it will errror

Reply:
> trim can't return null/undefined afaik

You are correct, but it can be an empty string and that's a falsey value. `compact()` removes only `undefined` and `null` values, which is why I had to return `null` for empty values.

Reply:
I missed the compact, usually in the codebase we do .filter(Boolean) which would achieve what you want and take care of the `""` I think.

```
["", null, undefined, "foo"].filter(Boolean) // ["foo"]
```

Anyways, up to you.

Reply:
Yes, but I would still have to the `trim` anyway. I think this is a matter of coding style, but it does not matter anymore because I removed this completely.